mllib/src/main/scala/org/apache/spark/ml/classification/DecisionTreeClassifier.scala

```diff
@@ -91,7 +91,7 @@ class DecisionTreeClassifier @Since("1.4.0") (
 
   /** @group setParam */
   @Since("1.4.0")
-  override def setImpurity(value: String): this.type = set(impurity, value)
+  def setImpurity(value: String): this.type = set(impurity, value)
```

Comment:
I think we can remove these too. I assumed they were meant to be deprecated because the trait's implementation was deprecated, and there's no compelling reason to only remove it from the trait. The point was to use set().

Reply:
I don't think those can be removed. The ones which were deprecated are the `setImpurity` on the `Model`s, not on the `Classifier`s/`Regressor`s which build the models.

Reply:
OK I think I'm wrong about the logic, given the first attempt to do this in https://github.com/apache/spark/pull/17867 Yes this is fine. In fact I think you're welcome to just revive that PR and remove a lot of similar deprecated methods again.

Reply:
I see, thanks. I'll update this PR removing all the other deprecated items for these models then. Thanks.